pmd-java/src/main/resources/category/java/errorprone.xml

```diff
@@ -1145,12 +1145,15 @@ public class JuniorClass extends SeniorClass {
           message="Found ''{0}''-anomaly for variable ''{1}'' (lines ''{2}''-''{3}'')."
           class="net.sourceforge.pmd.lang.java.rule.errorprone.DataflowAnomalyAnalysisRule"
           dfa="true"
+          deprecated="true"
           externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_errorprone.html#dataflowanomalyanalysis">
         <description>The dataflow analysis tracks local definitions, undefinitions and references to variables on different paths on the data flow.
 From those informations there can be found various problems.
 
 1. DU - Anomaly: A recently defined variable is undefined. These anomalies may appear in normal source text.
 2. DD - Anomaly: A recently defined variable is redefined. This is ominous but don't have to be a bug.
+
+This rule is deprecated. Use UnusedAssignment in category bestpractices instead.
```

Comment:
We can use here the rule tag as well, to make a link to the new rule. I'll change that when merging